client/classic/upe/index.js

```diff
@@ -110,16 +111,36 @@ jQuery( function ( $ ) {
 	let paymentIntentId = null;
 	let isUPEComplete = false;
 	const hiddenBillingFields = {
-		name: 'never',
-		email: 'never',
-		phone: 'never',
+		name:
+			enabledBillingFields.includes( 'billing_first_name' ) ||
+			enabledBillingFields.includes( 'billing_last_name' )
+				? 'never'
+				: 'auto',
+		email: enabledBillingFields.includes( 'billing_email' )
+			? 'never'
+			: 'auto',
+		phone: enabledBillingFields.includes( 'billing_phone' )
+			? 'never'
+			: 'auto',
 		address: {
-			country: 'never',
-			line1: 'never',
-			line2: 'never',
-			city: 'never',
-			state: 'never',
-			postalCode: 'never',
+			country: enabledBillingFields.includes( 'billing_country' )
+				? 'never'
+				: 'auto',
+			line1: enabledBillingFields.includes( 'billing_address_1' )
+				? 'never'
+				: 'auto',
+			line2: enabledBillingFields.includes( 'billing_address_2' )
+				? 'never'
+				: 'auto',
+			city: enabledBillingFields.includes( 'billing_city' )
+				? 'never'
+				: 'auto',
+			state: enabledBillingFields.includes( 'billing_state' )
+				? 'never'
+				: 'auto',
+			postalCode: enabledBillingFields.includes( 'billing_postcode' )
+				? 'never'
+				: 'auto',
```

Comment:
Now I'm concerned if we send `'-'` as postal code [below](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1795/files#diff-cafcea010e8e6c0d90e1dc3a72c2727bd09bb2f9038a5af2d3e625ca88e51ee3R260) that it will override whatever the customer enters in the stripe element. Could you tell by your testing if this was happening?

Reply:
Well I figured it out. Looking the the XHR requests in the network tab I found:
```
Request URL: https://api.stripe.com/v1/payment_intents/pi_3JXMFvB6usZchzRk0xzuo59b/confirm
Request Method: POST
Status Code: 200 
...
payment_method_data[billing_details][address][postal_code]: 83687
payment_method_data[type]: card
payment_method_data[card][number]: 4242 4242 4242 4242
payment_method_data[card][cvc]: 155
payment_method_data[card][exp_month]: 12
payment_method_data[card][exp_year]: 25
...
```

83687 is the zip code I entered in the stripe element so I think we're ok as it is.